contracts/MultiFeeDistributionFactory.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.12;
+
+import { MultiFeeDistribution } from "./MultiFeeDistribution.sol";
+import { IMultiFeeDistributionFactory } from "interfaces/IMultiFeeDistributionFactory.sol";
+import { IOwnable } from "interfaces/IOwnable.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+contract MultiFeeDistributionFactory is IMultiFeeDistributionFactory, Ownable {
+    bytes32 public override constant bytecodeHash =
+        keccak256(type(MultiFeeDistribution).creationCode);
+
+    // This called in the MultiFeeDistribution constructor
+    bytes public override cachedDeployData;
+
+    mapping(address => address) public override vaultToStaker;
+
+    constructor() {}
+
+    function deployStaker(address ichiVault) external override returns (address staker) {
+        bytes memory _deployData = abi.encode(ichiVault);
+        cachedDeployData = _deployData;
+
+        bytes32 salt = keccak256(_deployData);
+        staker = address(new MultiFeeDistribution{salt: salt}());
```

Comment:
That will prevent the 2nd deployment of a staker for the same vault, i.e. enforces 0,1:1, stakers:vaults

Reply:
Which is what we want!?

Reply:
Yes, I think so.